src/ray/raylet/scheduling/placement_group_location_index.h

```diff
@@ -0,0 +1,86 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "absl/container/flat_hash_map.h"
+#include "ray/common/id.h"
+#include "ray/common/placement_group.h"
+#include "ray/util/logging.h"
+
+namespace ray {
+
+class PlacementGroupLocationIndex {
```

Comment:
Isn't this almost the same class as `BundleLocationIndex` class? Can we consolidate these two?

Reply:
BundleLocationIndex has more BundleSpecification than PlacementGroupLocationIndex. It will be heavier.
The PlacementGroupLocationIndex I will then use for the following two optimizations:
1）Optimize the schedule performance of actors/tasks with pg for raylet schedule mode
2)  We consider removing the pg resource mark in resource synchronization. Thereby, the scheduling performance in large-scale clusters is improved and the pressure on pubsub is reduced.

At that time, the PlacementGroupLocationIndex needs to be synchronized in each node as a resource, and **the content needs to be minimized**, thereby reducing the size of the transmitted data. Improve performance.

Reply:
I thought about it again, and it is indeed better to conoslidate the bundle location index。I will spend a few days modifying the code to conoslidate the bundle location index.